src/XRP/issued-currency-client.ts

```diff
@@ -505,6 +505,32 @@ export default class IssuedCurrencyClient {
     )
   }
 
+  /**
+   * Disables rippling on the trust line between this account (issuing account) and another account.
+   * Note that the trust line's limit is set to 0.
+   *
+   * @see https://xrpl.org/freezes.html#enabling-or-disabling-individual-freeze
+   *
+   * @param trustLinePeerAccount The X-Address of the account involved in the trust line to disable rippling.
+   * @param currencyName The currency of the trust line to disable rippling.
+   * @amount The maximum amount of debt to allow on this trust line.
+   * @param wallet The wallet disabling rippling on the trust line.
+   */
+  public async setNoRippling(
+    trustLinePeerAccount: string,
+    currencyName: string,
+    amount: string,
```

Comment:
i found that if an existing trust line existed between the account and peer account, even if it was authorized, setting this to 0 would destroy this trust line. if, however, this was non-zero, then the `noRipple` would be properly set

Reply:
On some level it seems weird that it destroys the trustline... but I suppose that if you have a trustline that has 0 balance AND is frozen it's effectively useless and would need to be re-established by whoever has the power to unfreeze it anyway.

Reply:
Looking at the test... I wonder if a `noRipple` trustset transaction with a 0 limit would destroy the trustline if the counterparty had extended a trustline with a non-zero balance to the issuer?  I don't think it really matters for our purposes here since we can see that at least in the non-zero case the flag gets set.  The rest of this logic seems like rippled curiosities to me.